pkg/metallb/frr.go

```diff
@@ -0,0 +1,618 @@
+		"Collecting FRRConfiguration object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	unsObject, err := builder.apiClient.Resource(
+		GetFrrConfigurationGVR()).Namespace(builder.Definition.Namespace).Get(
+		context.TODO(), builder.Definition.Name, metav1.GetOptions{})
+
+	if err != nil {
+		glog.V(100).Infof(
+			"Failed to pull an unstructured object in namespace %s",
+			builder.Definition.Namespace)
+
+		return nil, err
+	}
+
+	return builder.convertToStructured(unsObject)
+}
+
+// Delete removes FrrConfigurationBuilder object from a cluster.
+func (builder *FrrConfigurationBuilder) Delete() (*FrrConfigurationBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Deleting the FrrConfiguration object %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace,
+	)
+
+	if !builder.Exists() {
+		return builder, fmt.Errorf("FrrConfiguration cannot be deleted because it does not exist")
+	}
+
+	err := builder.apiClient.Resource(
+		GetFrrConfigurationGVR()).Namespace(builder.Definition.Namespace).Delete(
+		context.TODO(), builder.Definition.Name, metav1.DeleteOptions{})
+
+	if err != nil {
+		return builder, fmt.Errorf("can not delete FRRConfiguration: %w", err)
+	}
+
+	builder.Object = nil
+
+	return builder, nil
+}
+
+// GetFrrConfigurationGVR returns FrrConfiguration GroupVersionResource which could be used for Clean function.
+func GetFrrConfigurationGVR() schema.GroupVersionResource {
+	return schema.GroupVersionResource{
+		Group: FRRAPIGroup, Version: APIVersion, Resource: "frrconfigurations",
+	}
+}
+
+// Exists checks whether the given FrrConfiguration exists.
+func (builder *FrrConfigurationBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof(
+		"Checking if FrrConfiguration %s exists in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// WithBGPRouter defines a BGP router as localASN.
+func (builder *FrrConfigurationBuilder) WithBGPRouter(localASN uint32, routerIndex int) *FrrConfigurationBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Define BGP router with an ASN: %v", localASN)
+
+	if strconv.Itoa(int(localASN)) == "" {
+		glog.V(100).Infof("The local ASN of the frrConfiguration can not be empty")
+
+		builder.errorMsg = "frrConfiguration 'local AS' cannot be empty"
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	// Ensure the routerIndex is within the bounds of the existing list
+	if routerIndex < 0 || routerIndex >= len(builder.Definition.Spec.BGP.Routers) {
+		// If the index is out of bounds, create a new router at the next available index
+		builder.Definition.Spec.BGP.Routers = append(builder.Definition.Spec.BGP.Routers, frrtypes.Router{
```

Comment:
Ok I changed the logic. It first check to see if the routerIndex is not greater then the routerIndex lenght. In this scenario 100 will fail. If the routerIndex is 0 it will create the first router if its not it will append.